providers/gcc.rb

```diff
@@ -26,11 +26,15 @@
     if new_resource.support_os.include?(os)
 #    if new_resource.support_os.include?($gecos_os)
 
-      gem_depends = [ 'rest_client' ]
+      gem_depends = [ 'rest-client' ]
+      gem_path = "/opt/chef/embedded/bin/gem"
```

Comment:
This "gem path search"  is used very often. I propose to move it to a generic place (like we did with "supported os detection") instead of repeating the same code in every provider.